### ISSUE TITLE
Truncate history

### ISSUE THREAD

**Issue by Timmmm** [gitx/gitx#151]
It would be nice if gitx had an option to not display all of history, e.g. only show the last 10k commits. If you open a huge repository, e.g. LLVM, then it is too slow to use nicely.

**Comment by tiennou** [comment]
That would be nice, but I fear it requires heavy changes to how the loading is done. Even though it's somewhat threaded now, having some kind of "scrolling triggers loading" would be too hard. I'll keep it open as an enhancement/pipe-dream though 😉.

**Comment by Timmmm** [comment]
Yeah I expected as much - probably will be easier when `libgit2` is used directly (assuming that is still the plan).

**Comment by tiennou** [comment]
We're already using libgit2 for building the graph (search `git_revwalk`/ `GTEnumerator`).

It's just that the API isn't built for it (and those calls sits below a few layers), so AFAIK it'll load everything every time. We would need some "would you kindly async-load the n commits that come after $ref ? kthxby", but then you encounter caching/threading and "hard things".

**Comment by rjmunro** [comment]
This sounded like it was addressing similar issues in git itself:
https://blogs.msdn.microsoft.com/devops/2018/06/25/supercharging-the-git-commit-graph/
